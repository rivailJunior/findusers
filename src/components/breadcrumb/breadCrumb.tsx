import React from 'react';
import styles from './breadCrumb.module.scss'

type Title = {
    onClick(): void,
    label: string,
    active?: boolean
}

type iBread = {
    titles: Title[]
}
const BreadCrumb = ({ titles }: iBread) => {
    return (
        <div className={styles.breadContainer}>
            {titles.map((bread, id) => {
                const isActive = bread.active ? `${ styles.bLabel } ${ styles.active }` : styles.bLabel;
                return <div data-testid="labelBread" key={id} className={isActive} onClick={bread.onClick}>{bread.label}</div>
            })}
        </div>
    );
}

export { BreadCrumb };
export default BreadCrumb;
