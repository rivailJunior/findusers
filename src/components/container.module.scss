.mainContainer {
    // position: absolute;
    display: grid;
    height: auto;

    // background-color: #eaeaead6;
    grid-template-columns: repeat(12, 1fr);
    align-self: center;
    margin-right: 15%;
    margin-left: 15%;
}

@mixin grid-column($columnSize) {
    grid-column: span $columnSize;
    min-height: 50px;
    padding: 10px;
}

$list: 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12;

@each $size in $list {
    .containerItem#{$size} {
        @include grid-column($size);
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .mainContainer {
        margin-right: 1%;
        margin-left: 1%;
    }
    @each $size in $list {
        .containerItem#{$size} {
            @include grid-column(12);
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}
