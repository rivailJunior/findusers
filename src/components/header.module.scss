.header {
    overflow: hidden;
    background-color: white;
    padding: 10px 10px;
}

.header a {
    float: left;
    color: black;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px;
}

.header a.logo {
    font-size: 25px;
    font-weight: bold;
}

.header a:hover {
    background-color: #ddd;
    color: black;
    cursor: pointer;
}

.header a.active {
    background-color: dodgerblue;
    color: white;
}

.header-right {
    float: right;
}

@media screen and (max-width: 500px) {
    .header a {
        float: none;
        display: block;
        text-align: left;
    }

    .header-right {
        float: none;
    }
}
