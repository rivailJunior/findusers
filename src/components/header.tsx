import React from 'react';
import styles from './header.module.scss'
import { SearchInput } from './searchInput';

type iHeader = {
    userName?: string,
    labelRight?: string,
    handleRight?(): Function
    handleSearch?: Function
}

const Header = ({ userName, labelRight, handleRight, handleSearch }: iHeader): JSX.Element => {
    return (
        <div className={styles.header}>
            <div className={styles.headerLeft}>
                <a href="#default" className={styles.logo}>{userName}</a>
            </div>
            <div className={styles.headerRight}>
                <div>
                    <SearchInput handleSearch={handleSearch} placeholder="Buscar Usuario" />
                    {labelRight && (
                        <div className={styles['header-right']}>
                            <a className={styles.active} onClick={handleRight} >Sair</a>
                        </div>
                    )}
                </div>

            </div>
        </div>

    );
}

export { Header };
