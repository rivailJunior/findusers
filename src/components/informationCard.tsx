import React from 'react';
import styles from './informationCard.module.scss';

type iCard = {
    title: string,
    mainText: string,

}
const InformationCard = ({ title, mainText }: iCard) => {
    return (
        <div className={styles.card}>
            <p>{title}</p>
            <h2>{mainText}</h2>
        </div>
    );
}

export { InformationCard };
