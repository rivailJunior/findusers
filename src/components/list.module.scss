@use '../pages/index.scss' as index;
@use '../pages/root.scss' as root;
.item {
    background-color: white;
    border: 0.5px solid #9e9e9e80;
    display: flex;
    flex-direction: column;
    overflow: hidden;
    border-radius: 5px;

    .itemLi {
        @include index.default-label(15px, 0, 100%, 300, 15px);
        display: grid;
        text-align: left;
        padding: 1%;
        display: grid;
        grid-template-columns: repeat(12, 1fr);
        max-height: 100px;

        .left {
            grid-column: span 2;
            display: flex;
            justify-content: center;
            align-self: center;
            font-size: 3rem;
        }
        .right {
            grid-column: span 10;

            .rightTitle {
                @include index.default-label(15px, 0, 100%, 700, 20px);
                // font-size: 1rem;
                text-transform: capitalize;
                text-align: left;
            }
            .rightSub {
                @include index.default-label(15px, 1%, 100%, 300, 20px, left);
                display: flex;
                flex-direction: column;
                justify-content: space-between;
            }
        }
    }
    .itemLi:hover {
        cursor: pointer;
        background-color: root.$hoverBgc;
        color: root.$hoverColor;
    }
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    .item {
        .itemLi {
            @include index.default-label(12px, 0, 100%, 500);

            grid-template-columns: repeat(12, 1fr);

            .left {
                display: none;
            }
            .right {
                grid-column: span 12;

                .rightTitle {
                    font-size: 1rem;
                    text-transform: capitalize;
                }
                .rightSub {
                    @include index.default-label(
                        10px,
                        1%,
                        100%,
                        300,
                        10px,
                        left
                    );
                    display: flex;
                    flex-direction: column;
                    justify-content: space-between;
                }
            }
        }
        .itemLi:hover {
            cursor: pointer;
            background-color: root.$hoverBgc;
            color: root.$hoverColor;
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
}

/* Desktops and laptops ----------- */
@media only screen and (min-width: 1224px) {
    /* Styles */
}
