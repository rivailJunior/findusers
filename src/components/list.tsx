import React from 'react';
import { ErrorFeedback } from './errorFeedback';
import styles from './list.module.scss';
import moment from 'moment'

type iList = {
    data: object[],
    handleClick: Function
}

const ListRepositories = ({ data, handleClick }: iList): JSX.Element => {
    return data.length ? (
        <div data-testid="list" className={styles.item}>
            {data.map((item: any, index) => {
                return (
                    <div key={index} className={styles.itemLi} onClick={() => handleClick(item)} data-testid="listItem">
                        <div className={styles.left}>
                            {item.stargazers_count}
                        </div>
                        <div className={styles.right}>
                            <div className={styles.rightTitle}>
                                {item?.name}
                            </div>
                            <div className={styles.rightSub}>
                                <div>{moment(item.created_at).format('DD/MM/YYYY')}</div>
                                <div> <a href={item.clone_url} target="_blank" rel="noopener noreferrer">{item.clone_url}</a></div>
                            </div>
                        </div>
                    </div>)
            })}
        </div>
    ) : (
        <ErrorFeedback title="Nada encontrado" />
    );
}

export { ListRepositories };
