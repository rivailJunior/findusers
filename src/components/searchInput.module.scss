@use '../pages/index.scss' as index;
.inputContainer {
    display: flex;
    flex-direction: row;

    .inputField {
        height: 50px;
        @include index.default-label(15px, 0, 100%, 300, 15px, left);
        border-radius: 5px;
        // border: none;
        border: thin solid #bec7d0;
        border-right: none;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    }

    .inputField:focus {
        border: dodgerblue;
        border-right: none;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    }

    button {
        background-color: dodgerblue;
        color: white;
        border: none;
        border-radius: 5px;
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
    }

    .disabled {
        background-color: #bec7d0;
    }
}
