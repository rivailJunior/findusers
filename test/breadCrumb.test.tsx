import React from 'react'
import { unmountComponentAtNode } from "react-dom";
import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { BreadCrumb } from '../src/components/breadCrumb'

let container = null;
beforeEach(() => {
    // configurar o elemento do DOM como o alvo da renderização
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // limpar na saída
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});
const fn = jest.fn()
const breadCrumb = [{ label: 'Descrição usuário/', onClick: fn, active: true }, { label: 'Detalhes repositório/', onClick: () => null }]

describe('BreadCrumb', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<BreadCrumb titles={breadCrumb} />)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Show more than one value', () => {
        render(<BreadCrumb titles={breadCrumb} />, container)
        expect(screen.getByText(/Descrição usuário/i)).toBeInTheDocument();
        expect(screen.getByText(/Detalhes repositório/i)).toBeInTheDocument();
    })

    test('Show one of than with active class', () => {
        render(<BreadCrumb titles={breadCrumb} />, container)
        const getItems = screen.getAllByTestId('labelBread')
        expect(getItems[0]).toHaveClass('active');
        expect(getItems[1]).not.toHaveClass('active');
    })

    test('Handle Click on ative item', () => {

        render(<BreadCrumb titles={breadCrumb} />, container)
        const getItems = screen.getAllByTestId('labelBread')
        fireEvent.click(getItems[0])
        expect(fn).toHaveBeenCalledTimes(1);
    })
})