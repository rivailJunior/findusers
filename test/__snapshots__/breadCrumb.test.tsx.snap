// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Render Correctly 1`] = `
<div
  className="breadContainer"
>
  <div
    className="bLabel active"
    data-testid="labelBread"
    onClick={[MockFunction]}
  >
    Descrição usuário/
  </div>
  <div
    className="bLabel"
    data-testid="labelBread"
    onClick={[Function]}
  >
    Detalhes repositório/
  </div>
</div>
`;
